addons/mail/models/discuss/discuss_channel.py

```diff
@@ -160,6 +161,17 @@ def _constraint_group_id_channel(self):
 
     # COMPUTE / INVERSE
 
+    @api.depends("channel_member_ids", "auto_recompute_name")
+    def _compute_name(self):
+        for channel in self:
+            if channel.auto_recompute_name:
+                partners_to_name = channel.channel_member_ids.partner_id[:5]
+                new_name = partners_to_name[0].name if partners_to_name else ""
+                if len(partners_to_name) > 1:
+                    new_name = ', '.join(partners_to_name[0:-1].mapped('name')) + ' and ' + partners_to_name[-1].name
+                if channel.name != new_name:
+                    channel.write({'name': new_name})
```

Comment:
compute should assign value not use the write method (this won't work for NewId, and create other kinds of bugs)